src/snowflake/connector/query_context_cache.py

```diff
@@ -0,0 +1,284 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2012-2021 Snowflake Computing Inc. All rights reserved.
+#
+from __future__ import annotations
+
+from base64 import b64decode, b64encode
+from functools import total_ordering
+from hashlib import md5
+from io import BytesIO
+from logging import DEBUG, getLogger
+from threading import Lock
+
+from sortedcontainers import SortedSet
+
+from .options import installed_pandas
+from .options import pyarrow as pa
+
+logger = getLogger(__name__)
+
+
+if installed_pandas:
```

Comment:
why dependency on Pandas?

Reply:
`pyarrow` is installed with snowflake connector along with pandas therefore we use this check